### ISSUE TITLE
[no-unsafe-call][no-unsafe-member-access] False positives on newer static methods

### ISSUE THREAD

**Issue by philipbordallo** [typescript-eslint/typescript-eslint#2395]
Some newer static methods are causing false positives for `no-unsafe-call` & `no-unsafe-member-access` in situations where they shouldn't be.
This includes:
- [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from) – `no-unsafe-call`
- [`Array.of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/of) – `no-unsafe-call`
- [`Object.entries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) – `no-unsafe-call`
- [`Object.fromEntries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries) – `no-unsafe-call`
- [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) – `no-unsafe-call`
- [`Object.values`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values) – `no-unsafe-call`
- [`Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) – `no-unsafe-call` & `no-unsafe-member-access`
- [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) – `no-unsafe-call`  & `no-unsafe-member-access`
- [`Promise.race`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race) – `no-unsafe-call` & `no-unsafe-member-access`
- [`Promise.resolve`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve) – `no-unsafe-call` & `no-unsafe-member-access`
- [`Promise.reject`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject) – `no-unsafe-call` & `no-unsafe-member-access`

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-call": "error",
    "@typescript-eslint/no-unsafe-member-access": "error"
  }
}
```

```TS
Array.from([1]);
Array.of([2]);

Object.values({ key: 'value' });
Object.entries({ key: 'value' });
Object.fromEntries([['key', 'value']]);
Object.is('one', 'one');

Promise.allSettled([
  Promise.resolve(1),
  Promise.reject(),
]);

Promise.all([
  Promise.resolve(1),
  Promise.reject(),
]);

Promise.race([
  Promise.resolve(1),
  Promise.reject(),
]);
```

```json
{
  "compilerOptions": {
    "module": "commonjs",
    "lib": [
      "ES2020"
    ]
  },
  "exclude": [
    "node_modules"
  ]
}

```

**Expected Result**

No errors.

**Actual Result**

- 15 `no-unsafe-call` errors
- 9 `no-unsafe-member-access` errors

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.9.0` |
| `@typescript-eslint/parser`        | `3.9.0` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.7.0` |
| `node`                             | `14.7.0` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
I'm unable to repro this.
Could you please create an isolated repro repo that I can checkout to help debug?

**Comment by philipbordallo** [comment]
@bradzacher it looks like I'm having you chase ghosts as I created a isolated repo and wasn't able to reproduce either. 

When I went back to the VSCode workspace where I first noticed this issue with `Array.from` and `Object.entries`, it was still showing the error but then a restart of the TypeScript server cleared it all up. And no combination of configuration either from ESLint or TypeScript seems to bring it back up.

I feel like we can close this issue, apologies for an inconvenience and appreciate the quick response!
